### PR TITLE
Channel cleanups

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Some simple cleanups I came across while starting work on #138.

### CODE REVIEW

File: src/ln/channel.rs [hunk 1]
```diff
@@ -338,18 +338,17 @@ const B_OUTPUT_PLUS_SPENDING_INPUT_WEIGHT: u64 = 104; // prevout: 40, nSequence:
 pub const MAX_FUNDING_SATOSHIS: u64 = (1 << 24);
 
 macro_rules! secp_call {
-	( $res: expr, $err: expr ) => {
+	( $res: expr, $err: expr, $chan_id: expr ) => {
 		match $res {
 			Ok(key) => key,
-			//TODO: make the error a parameter
-			Err(_) => return Err(HandleError{err: $err, action: Some(msgs::ErrorAction::DisconnectPeer{ msg: None })})
+			Err(_) => return Err(HandleError {err: $err, action: Some(msgs::ErrorAction::SendErrorMessage {msg: msgs::ErrorMessage {channel_id: $chan_id, data: $err.to_string()}})})
```
Comment (ariard) [inline src/ln/channel.rs]: One more #153 case isn't it ?

File: src/ln/channelmanager.rs [hunk 2]
```diff
@@ -1147,7 +1147,12 @@ impl ChannelManager {
...
 						let (commitment_msg, monitor) = match forward_chan.send_commitment() {
 							Ok(res) => res,
-							Err(_e) => {
+							Err(e) => {
+								if let &Some(msgs::ErrorAction::DisconnectPeer{msg: Some(ref _err_msg)}) = &e.action {
+								} else if let &Some(msgs::ErrorAction::SendErrorMessage{msg: ref _err_msg}) = &e.action {
+								} else {
+									panic!("Stated return value requirements in send_commitment() were not met");
+								}
 								//TODO: Handle...this is bad!
```
Comment (ariard) [inline src/ln/channelmanager.rs]: seems all kind of errors are covered by the last else, so TODO can disappear?

**Comment by ariard** [comment]
Concept ACK, but to be honest doesn't have (yet) all the subtilities of HTLCState in head!

File: src/ln/channel.rs (hunk 1 above)
Comment (TheBlueMatt) [reply src/ln/channel.rs]: Yea, I'm anticipating the fix for #153 is to just remove the distinction, have a general ChannelError type, then convert it to a HandleError in ChannelManager. That would simplify some of the Internal error awkwardness right now anyway.

File: src/ln/channelmanager.rs (hunk 2 above)
Comment (TheBlueMatt) [reply src/ln/channelmanager.rs]: Heh, the ifs/panic is just to give the fuzz tester something to attack. The actually *handling* still has to happen.

**Pull request merged**